Watchman.Discord.UnitTests/Users/UsersControllerTests.cs

```diff
@@ -101,5 +105,53 @@ public async Task GetAvatar_ShouldDisplayUserOwnAvatar()
             messagesServiceMock.Verify(x => x.SendMessage(contexts.User.AvatarUrl, It.IsAny<MessageType>()), Times.Once);
             messagesServiceMock.Verify(x => x.SendResponse(It.IsAny<Func<IResponsesService, string>>()), Times.Never);
         }
+        [Test, AutoData]
+        public async Task AddRole_ShouldAddRole(AddRoleCommand command)
+        {
+            //Arrange
+            var contexts = testContextsFactory.CreateContexts(1, 1, 1, "test");
+            var safeRole = new SafeRole(1ul, 3ul);
+            var safeRoles = new List<SafeRole>();
+            safeRoles.Add(safeRole);
+
+            var rolesServiceMock = new Mock<Areas.Users.Services.IRolesService>();
+            var queryBusMock = new Mock<IQueryBus>();
+            queryBusMock.Setup(x => x.Execute(It.IsAny<GetDiscordServerSafeRolesQuery>()))
+                .Returns(new GetDiscordServerSafeRolesQueryResult(safeRoles));
+
+            var controller = this.testControllersFactory.CreateUsersController(
+                queryBusMock: queryBusMock,
+                rolesServiceMock: rolesServiceMock);
+            
+            //Act
+            await controller.AddRole(command, contexts);
+
+            //Assert
+            queryBusMock.Verify(x => x.Execute(It.IsAny<GetDiscordServerSafeRolesQuery>()), Times.Once);
+            rolesServiceMock.Verify(x => x.AddRoleToUser(It.IsAny<IEnumerable<SafeRole>>(), It.IsAny<Contexts>(), It.IsAny<List<string>>()), Times.Once);
+        }
+        [Test, AutoData]
+        public async Task AddRole_ShouldThrowExceptionIfRulesCountIsGreaterThanFive(AddRoleCommand command)
+        {
+            //Arrange
+            var contexts = testContextsFactory.CreateContexts(1, 1, 1, "test");
+            command.Roles.Add("test");
```

Comment:
zrób sobie tą komende normalnie a nie przez autodata

tu masz przypadek w którym to jakie masz dane, ma znaczenie